src/libs/Navigation/AppNavigator/index.tsx

```diff
@@ -10,6 +12,16 @@ type AppNavigatorProps = {
 };
 
 function AppNavigator({authenticated}: AppNavigatorProps) {
+    const currentReportIDValue = useCurrentReportID();
+    const currentReportID = useNavigationState((state) => currentReportIDValue?.getCurrentReportID(state));
+
+    useEffect(() => {
+        currentReportIDValue?.updateCurrentReportID(currentReportID);
+
+        // eslint-disable-next-line react-compiler/react-compiler
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we only want to run the effect on currentReportID change.
+    }, [currentReportID]);
```

Comment:
Why move this code from the NavigationRoot to the AppNavigator? Does this fix not work without this change?